### ISSUE TITLE
Neon should warn/throw when the `bundleName` parameter is incorrectly omitted from a `LanguageConfiguration` definition.

### ISSUE THREAD

**Issue by dvclmn** [ChimeHQ/Neon#49]
I had the below example snippet in my `makeHighlighter` function, and was scratching my head when inline markdown (bold, italic, strikethrough, inline code, etc) was not being parsed/highlighted/ returned via `let provider: TokenAttributeProvider = { token in`.

```swift
static func makeHighlighter(for textView: MarkdownTextView) throws -> TextViewHighlighter {

  let markdownConfig = try LanguageConfiguration(
    tree_sitter_markdown(),
    name: "Markdown"
  )
  let markdownInlineConfig = try LanguageConfiguration(
    tree_sitter_markdown_inline(),
    name: "Markdown Inline"
  )
```

Turns out I had left out the `bundleName` parameter from `let markdownInlineConfig = try LanguageConfiguration` — this definition should have looked like:

```swift
let markdownInlineConfig = try LanguageConfiguration(
  tree_sitter_markdown_inline(),
  name: "Markdown Inline",
  bundleName: "TreeSitterMarkdown_TreeSitterMarkdownInline"
)
```

It'd be ideal if Neon could catch this issue, and throw an error, to save some debugging time.

To quote yourself, to yourself, for posterity:
> I think probably what has to happen is when it executes a deep query, cross the entire document tree and when it encounters a language that does not have that query defined, it should complain

**Comment by mattmassicotte** [comment]
This is tricky to fix, because highlighting is intentionally a "no failure permitted" operation. A `TokenProvider` cannot throw, because it's unclear what a `RangeValidator` should even do no failure. But, first things first. SwifTreeSitter wasn't correctly throwing if a query was executed that could not be satisified. That was easy to expose with a test and fix.

https://github.com/ChimeHQ/SwiftTreeSitter/commit/b31cf6fe94238db0388edf74c3482b5b6a0c16bb

**Comment by mattmassicotte** [comment]
@dvclmn what do you think? If the system attemps to get styles for a range and fails, what do you think should happen? In the past, I've used a delay+retry. This worked poorly because not all errors are transient...

**Comment by mattmassicotte** [comment]
Ok, I wasn't thinking clearly here. It makes no sense to be able to initialize a `LanguageConfiguration` with a bundle that does not exist. There are other initializers available - using a bundle is not a requirement. This is a SwiftTreeSitter bug and has been fixed.

https://github.com/ChimeHQ/SwiftTreeSitter/commit/d0938d3607abcc5669bdf1d7d3b0485a8740fc65

**Comment by mattmassicotte** [comment]
Fixed with 8d01febc54d0bedba1136ef3bc833ddd7c843528
